pkg/localstore/sampler.go

```diff
@@ -104,6 +112,13 @@ func (db *DB) ReserveSample(ctx context.Context, anchor []byte, storageDepth uin
 					continue
 				}
 
+				// check if the timestamp on the postage stamp is not later than
+				// the consensus time.
+				if time.Duration(binary.BigEndian.Uint64(chItem.Timestamp))*time.Nanosecond > consensusTime {
```

Comment:
doesnt this lead to an extra 2-4 million conversions to time.Duration?

Reply:
also if time.duration is an alias for int64 isn't this conversion theoretically unsafe?

Reply:
Storing timestamps with int 64 bits is safe for decades to come, so no danger here.